### PR TITLE
DDF-1816: Update ConfigurationMigrationManager to call Migratable services

### PR DESCRIPTION
**Pull request by lessarderic**
Updated ConfigurationMigrationManager and added unit tests.

Reviewers:
@andrewkfiedler 
@garrettfreibott 
@figliold 
@Lambeaux 
@oconnormi

### CODE REVIEW

File: platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/ConfigurationMigrationManager.java [hunk 1]
```diff
@@ -100,6 +108,7 @@ public void init() throws Exception {
             Files.createDirectories(exportDirectory);
             configurationAdminMigration.export(exportDirectory);
             migrationWarnings.addAll(systemConfigurationMigration.export(exportDirectory));
```
Comment (figliold) [inline platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/ConfigurationMigrationManager.java]: We are going to remove the line above after we get all of the Migratables finished..right?

File: platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/ConfigurationMigrationManager.java [hunk 2]
```diff
@@ -119,4 +129,34 @@ public void init() throws Exception {
...
+
+    private Collection<MigrationWarning> exportMigratables(Path exportDirectory) {
+        List<MigrationWarning> warnings = new LinkedList<>();
+
+        for (ServiceReference<Migratable> serviceReference : getMigratableServiceReferences()) {
+
+            Migratable migratable = getBundleContext().getService(serviceReference);
+            MigrationMetadata migrationMetadata =
+                    migratable.export(exportDirectory.resolve((String) serviceReference.getProperty(
+                            "service.pid")));
```
Comment (figliold) [inline platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/ConfigurationMigrationManager.java]: Should we use a constant for service.pid? https://osgi.org/javadoc/r2/org/osgi/framework/Constants.html#SERVICE_PID

File: platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/ConfigurationMigrationManager.java (hunk 1 above)
Comment (lessarderic) [inline platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/ConfigurationMigrationManager.java]: Yep.

File: platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/ConfigurationMigrationManager.java (hunk 2 above)
Comment (lessarderic) [inline platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/ConfigurationMigrationManager.java]: Great catch! Will fix.

File: platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/ConfigurationMigrationManager.java [hunk 3]
```diff
@@ -119,4 +131,34 @@ public void init() throws Exception {
...
+        }
+
+        return warnings;
+    }
+
+    private Collection<ServiceReference<Migratable>> getMigratableServiceReferences() {
+        try {
+            return getBundleContext().getServiceReferences(Migratable.class, null);
+        } catch (InvalidSyntaxException e) {
+            String message = "Failure to export, internal error occurred.";
```
Comment (Lambeaux) [inline platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/ConfigurationMigrationManager.java]: Is there anyway to be more specific with this error message? I know we don't want to expose implementation details to the user, but if I saw this error while debugging (which is used in multiple places), I wouldn't know where to go to solve it.

Comment (lessarderic) [inline platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/ConfigurationMigrationManager.java]: Changed log message. Exception and stack trace will also get logged which should help too.

**Comment by Lambeaux** [comment]
:+1: :snowman: :8ball: :weary: :sunflower: :star: :statue_of_liberty: :bar_chart:

**Comment by lessarderic** [comment]
All builds successful and heroing completed. Merging.

**Pull request merged**